CDCgov/prime-reportstream
# Update HL7 Validation Error Messaging

This PR uses the document Seth created from #6307 to update the back-end code with new error reporting messages. Please use the utilities included in the attached ticket #5643 to test this on your local machine.

Test Steps:
1. Pull latest on this branch and spin up the Docker containers, RS Router, and Postman
2. Use Postman to send a message to `http://localhost:7071/api/validate` with any of the provided error-laden files
3. Ensure the messages presented match the desired messages in the spreadsheet from Seth and O&O.

## Changes
- Updates to error reporting messages for file validation

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #5643 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
